### PR TITLE
Switch inferred spans tests to pytest parametrized.

### PR DESCRIPTION
**Pull request by purple4reina**
### What does this PR do?

Uses `pytest.mark.parametrize` to create a single parameterized test for creating inferred spans.

### Motivation

Cleaner. Easier to read. <3 pytest

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

https://datadoghq.atlassian.net/browse/SVLS-4291

### CODE REVIEW

File: tests/test_tracing.py [hunk 1]
```diff
@@ -48,32 +49,6 @@
...
-
-
-def wrapped_span_creator(span_creator_func):
-    def result_func(*args, **kwargs):
-        return register_span(span_creator_func(*args, **kwargs))
-
-    return result_func
-
-
-create_inferred_span = wrapped_span_creator(create_inferred_span)
```
Comment (purple4reina) [inline tests/test_tracing.py:75]: Tests still pass if I remove this stuff

File: tests/test_tracing.py [hunk 2]
```diff
@@ -1369,545 +1190,589 @@ def test_remaps_specific_inferred_span_service_names_from_eventbridge_event(
...
+    if expect.parent_name is not None:
+        assert mock_span_finish.call_count == 1
+        args, kwargs = mock_span_finish.call_args_list[0]
+        parent = args[0]
+        finish_time = kwargs.get("finish_time") or args[1]
+        assert parent.name == expect.parent_name
+        assert actual.parent_id == parent.span_id
+        assert finish_time == expect.start
+    else:
+        assert mock_span_finish.call_count == 0
```
[the remark this answers is not in the document]
Comment (purple4reina) [reply tests/test_tracing.py:1772]: Nope! Maybe I should change this, but I overloaded `parent_name`. If set, then we essentially assume there's two inferred spans. If not set, then there's only one inferred span.

[the remark this answers is not in the document]
Comment (purple4reina) [reply tests/test_tracing.py:1772]: I'll add a comment for clarification

[the remark this answers is not in the document]
Comment (duncanista) [reply tests/test_tracing.py:1772]: Perfect

**Review by duncanista — APPROVED** [review]

**Review by joeyzhao2018 — COMMENTED** [review]
👍

**Review by astuyve — APPROVED** [review]

**Pull request merged**